doc/design/software-bridge-management.md

```diff
@@ -0,0 +1,495 @@
+}
+
+// LinuxBrConfig optional configuration for Linux bridge and uplink interface
+type LinuxBrConfig struct {
+	Interface map[string]string    // TODO clarify required settings
+	Bridge    *LinuxBrBridgeConfig `json:"bridge,omitempty"`
+}
+
+// LinuxBrBridgeConfig contains some options for linux bridge
+type LinuxBrBridgeConfig struct {
+	VlanFiltering bool `json:"vlan_filtering,omitempty"`
+	// +kubebuilder:validation:Enum=802.1Q;802.1ad
+	VlanProtocol string `json:"vlan_protocol,omitempty"`
+}
+```
+
+Open questions:
+- do we want to support each OVS Bridge/Interface option explicitly or should we support
+ generic map[string]string struct which allows set any field in the Bridge/Interface tables?
+- need to clarify list of required args for OVS and Linux bridge
+
+_**Note 1**: multiple policies can match single PF (vf range use-case), bridge settings from the policy with
+ the higher priority (which is the one with the lowest `.spec.priority` value) will be applied to PF._
+
+_**Note 2**: multiple NICs can match the same policy on a host. In this case a separate bridge will be created for each NIC._
+
+#### SriovNetworkNodeState CRD
+
+```golang
+type Interface struct {
+	PciAddress        string    `json:"pciAddress"`
+	NumVfs            int       `json:"numVfs,omitempty"`
+	Mtu               int       `json:"mtu,omitempty"`
+	Name              string    `json:"name,omitempty"`
+	LinkType          string    `json:"linkType,omitempty"`
+	EswitchMode       string    `json:"eSwitchMode,omitempty"`
+	VfGroups          []VfGroup `json:"vfGroups,omitempty"`
+	ExternallyManaged bool      `json:"externallyManaged,omitempty"`
+	Bridge            *Bridge   `json:"bridge,omitempty"`
+}
+
+
+type InterfaceExt struct {
+	Name              string            `json:"name,omitempty"`
+	Mac               string            `json:"mac,omitempty"`
+	Driver            string            `json:"driver,omitempty"`
+	PciAddress        string            `json:"pciAddress"`
+	Vendor            string            `json:"vendor,omitempty"`
+	DeviceID          string            `json:"deviceID,omitempty"`
+	NetFilter         string            `json:"netFilter,omitempty"`
+	Mtu               int               `json:"mtu,omitempty"`
+	NumVfs            int               `json:"numVfs,omitempty"`
+	LinkSpeed         string            `json:"linkSpeed,omitempty"`
+	LinkType          string            `json:"linkType,omitempty"`
+	EswitchMode       string            `json:"eSwitchMode,omitempty"`
+	ExternallyManaged bool              `json:"externallyManaged,omitempty"`
+	TotalVfs          int               `json:"totalvfs,omitempty"`
+	VFs               []VirtualFunction `json:"Vfs,omitempty"`
+	Bridge            *Bridge           `json:"bridge,omitempty"`
+}
+
+```
+`SriovNetworkNodeState.spec` and `SriovNetworkNodeState.status` should be extended to contain same `Bridge` struct as `SriovNetworkNodePolicySpec` CRD. 
```

Comment:
I don't think we can use the same object as the Policy as in the policy I can have nicSelector: vendor: 8086 
but in the spec and status for the NodeState I will like to have a list of all the bridges we need to create with the configuration, name and uplink object

Reply:
@SchSeba I updated proposed API change for SriovNetworkNodeState object. Now it contains Bridges list.
I propose to use the same Bridges object in the spec and in the status. We may want to extend the one in the status field with additional info when we will have feedback from the users.